tests/models/test_openai.py

```diff
@@ -3085,3 +3087,146 @@ async def test_cache_point_filtering_responses_model():
+                    timestamp=IsNow(tz=timezone.utc),
+                ),
+            ],
+            run_id=IsStr(),
+        )
+    ]
+
+    chat_model = OpenAIChatModel(model_name, provider=OpenAIProvider(api_key='foobar'))
+    usage_result: RequestUsage = await chat_model.count_tokens(test_messages, {}, ModelRequestParameters())
+    assert usage_result.input_tokens == expected_token_count
+
+    responses_model = OpenAIResponsesModel(model_name, provider=OpenAIProvider(api_key='foobar'))
+    usage_result: RequestUsage = await responses_model.count_tokens(test_messages, {}, ModelRequestParameters())
+    assert usage_result.input_tokens == expected_token_count
+
+
+@pytest.mark.vcr()
+async def test_count_tokens_with_non_string_values():
+    """Test token counting with messages that have non-string values (like content arrays)."""
+    test_messages: list[ModelMessage] = [
+        ModelRequest(
+            parts=[
+                UserPromptPart(
+                    content=[
+                        'Text content',
+                        ImageUrl(url='https://example.com/image.jpg'),
+                    ],
+                    timestamp=IsNow(tz=timezone.utc),
+                ),
+            ],
+            run_id=IsStr(),
+        )
+    ]
+
+    chat_model = OpenAIChatModel('gpt-4o', provider=OpenAIProvider(api_key='foobar'))
+    usage_result: RequestUsage = await chat_model.count_tokens(test_messages, {}, ModelRequestParameters())
+
+    assert usage_result.input_tokens > 0
+
+
+@pytest.mark.vcr()
+async def test_openai_model_usage_limit_not_exceeded(
+    allow_model_requests: None,
+    openai_api_key: str,
+):
+    provider = OpenAIProvider(api_key=openai_api_key)
+    model = OpenAIResponsesModel('gpt-4', provider=provider)
+    agent = Agent(model=model)
+
+    result = await agent.run(
+        'The quick brown fox jumps over the lazydog.',
+        usage_limits=UsageLimits(input_tokens_limit=25, count_tokens_before_request=True),
+    )
+    assert result.output == snapshot(
+        "That sentence is a pangram, meaning it contains every letter of the English alphabet. It's often used to test "
+        'fonts or demonstrate handwriting, because it includes all the letters.'
+    )
+
+
+@pytest.mark.vcr()
+async def test_openai_model_usage_limit_exceeded(
+    allow_model_requests: None,
+    openai_api_key: str,
+):
+    provider = OpenAIProvider(api_key=openai_api_key)
+    model = OpenAIResponsesModel('gpt-4', provider=provider)
+    agent = Agent(model=model)
+
+    with pytest.raises(UsageLimitExceeded) as exc_info:
+        _ = await agent.run(
+            'The quick brown fox jumps over the lazydog. The quick brown fox jumps over the lazydog.',
+            usage_limits=UsageLimits(input_tokens_limit=25, count_tokens_before_request=True),
+        )
+
+    assert 'exceed the input_tokens_limit of' in str(exc_info.value)
```

Comment:
Please test the entire exception message as here:

https://github.com/pydantic/pydantic-ai/blob/c05059040ea9d44f3a1df8200f2c5735d82ea499/tests/models/test_anthropic.py#L6449-L6464